### PR TITLE
Refact: huge server refactoring

### PR DESCRIPTION
**Pull request by thecampagnards**

### CODE REVIEW

File: server/handler/daemons/cadvisor.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	"docktor/server/dao"
+	"docktor/server/types"
+	"docktor/server/utils"
+
+	"github.com/labstack/echo"
+	log "github.com/sirupsen/logrus"
+)
+
+// getCAdvisorContainerInfo get cadvisor info of container
+func getCAdvisorContainerInfo(c echo.Context) error {
```
Comment (thecampagnards) [inline server/handler/daemons/cadvisor.go]: revoir a part d'ici

File: server/handler/groups/group.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		return c.JSON(http.StatusBadRequest, err.Error())
+	}
+	daemons, err := dao.GetDaemons()
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error": err,
+		}).Error("Error when retrieving daemons")
+		return c.JSON(http.StatusBadRequest, err.Error())
+	}
+	for key, group := range groups {
```
Comment (thecampagnards) [inline server/handler/groups/group.go]: Do it in dao

Comment (thecampagnards) [reply server/handler/groups/group.go]: remove duplication https://sonarcloud.io/component_measures?id=thecampagnards_docktor&metric=new_duplicated_lines_density&pullRequest=27&selected=thecampagnards_docktor%3Aserver%2Futils%2Fcompose.go

Comment (thecampagnards) [reply server/handler/groups/group.go]: add unit test for admin

Comment (SmalkiGroove) [reply server/handler/groups/group.go]: Tu parles tout seul mec ?

Comment (thecampagnards) [reply server/handler/groups/group.go]: @SmalkiGroove yep xD

**Pull request merged**